projects/graph/src/graph_demo.py

```diff
@@ -5,12 +5,86 @@
 """
 
 from sys import argv
+from graph import Graph
+from draw import BokehGraph
+import random
 
 
-def main():
-    pass  # TODO
+def createDefaultGraph():
+    graph = Graph()  # Instantiate your graph
+    graph.add_vertex('0')
+    graph.add_vertex('1')
+    graph.add_vertex('2')
+    graph.add_vertex('3')
+    graph.add_edge('0', '1')
+    graph.add_edge('0', '3')
+
+    bg = BokehGraph(graph)
+    bg.draw()
+
+
+# O(n^2)
+def createRandomGraph(numNodes, numEdges):
+    graph = Graph()  # Instantiate your graph
+
+    all_edges = []
+
+    # O(n^2)
+    for i in range(numNodes):
+        for j in range(i + 1, numNodes):
+            all_edges.append((i,  j))
+
+    # O(n)
+    random.shuffle(all_edges)
+    edges = all_edges[:numEdges]
+
+    # O(1)
+    if numEdges > len(all_edges):
+        print("Warning: Too many edges")
+
+    # O(n)
+    for edge in edges:
+        print(edge)
+
+    # O(n)
+    for i in range(numNodes):
+        graph.add_vertex(i)
+
+    # O(n)
+    for edge in edges:
+        print(edge)
+        graph.add_edge(edge[0], edge[1])
+
+    print(len(edges))
+
+    bg = BokehGraph(graph)
+    bg.draw()
+
```

Comment:
I  think this would work if you had a draw method in your BokehGraph class (in draw.py).  Since there is no separate method for drawing you would need to tell it the output file and plot manually.